### PR TITLE
Remove AnalyzerExecutor.CancellationToken property

### LINKED ISSUE #67670: Remove AnalyzerExecutor.CancellationToken property
Follow-up refactoring to avoid bugs such as https://github.com/dotnet/roslyn/issues/67084

Instead of a property on this type, `cancellationToken` should be a method parameter and threaded through all the method calls within this type.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #67670
Follow-up refactoring to avoid bugs such as #67084

Instead of a property on this type, cancellationToken is now threaded through all the method calls as a parameter.

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs [hunk 1]
```diff
@@ -1084,7 +1084,7 @@ private static IEnumerable<Diagnostic> GetEffectiveDiagnosticsImpl(ImmutableArra
             Action<Exception, DiagnosticAnalyzer, Diagnostic, CancellationToken>? wrappedOnAnalyzerException =
                 onAnalyzerException == null ? null : (ex, analyzer, diagnostic, _) => onAnalyzerException(ex, analyzer, diagnostic);
             var analyzerExecutor = AnalyzerExecutor.CreateForSupportedDiagnostics(wrappedOnAnalyzerException, analyzerManager);
-            return AnalyzerDriver.IsDiagnosticAnalyzerSuppressed(analyzer, options, analyzerManager, analyzerExecutor, analysisScope: null, severityFilter: SeverityFilter.None);
+            return AnalyzerDriver.IsDiagnosticAnalyzerSuppressed(analyzer, options, analyzerManager, analyzerExecutor, analysisScope: null, severityFilter: SeverityFilter.None, CancellationToken.None);
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs:1087]: >CancellationToken.None [](http://example.com/codeflow?start=177&length=22)

Is it fine to pass None here?

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs [hunk 2]
```diff
@@ -523,7 +523,7 @@ public bool TryGetValue<TValue>(SyntaxTree tree, SyntaxTreeValueProvider<TValue>
         private readonly Compilation _compilation;
         private readonly AnalyzerOptions _options;
         private readonly Action<Diagnostic> _reportDiagnostic;
-        private readonly Func<Diagnostic, bool> _isSupportedDiagnostic;
+        private readonly Func<Diagnostic, CancellationToken, bool> _isSupportedDiagnostic;
         private readonly CompilationAnalysisValueProviderFactory? _compilationAnalysisValueProviderFactoryOpt;
         private readonly CancellationToken _cancellationToken;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs:528]: >private readonly CancellationToken _cancellationToken; [](http://example.com/codeflow?start=8&length=54)

Is it fine to have this field?

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1)

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs (hunk 1 above)
Comment (mavasani) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs:1087]: This public API doesn’t have a CancellationToken parameter, so we don’t have any option. There is also a proposal to deprecate this public API: https://github.com/dotnet/roslyn/issues/67592

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs (hunk 2 above)
Comment (mavasani) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs:528]: This is exposed as a public property on every analysis context type, so yes this is intentional

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Pull request merged**